### PR TITLE
Adding key-vault related configs for Azure

### PR DESCRIPTION
**Pull request by manikarthikk**
#### What does this PR do? Please provide some context
Adding key-vault related configs for Azure
The following configs will be added defaulting to empty string
```

In lms.env.json: 

    "AZURE_MSI_REQUEST_URI" : "http://169.254.169.254/metadata/identity/oauth2/token",
    "GET_SECRETS_FROM_AZURE_KEYVAULT": true,
    "KEYVAULT_PAYLOAD":  {"resource": "https://vault.azure.net","api-version": "2018-02-01"},
    "KEYVAULT_URL" : "https://{keyvaultname}.vault.azure.net",
    "LMS_NON_SECURE_KEY_NAME": {key name from keyvault},
    "LMS_SECURE_KEY_NAME": {key name from keyvault },
    "API_VERSION": "2016-10-01"

In cms.env.json: 

    "AZURE_MSI_REQUEST_URI" : "http://169.254.169.254/metadata/identity/oauth2/token",
    "GET_SECRETS_FROM_AZURE_KEYVAULT": true,
    "KEYVAULT_PAYLOAD":  {"resource": "https://vault.azure.net","api-version": "2018-02-01"},
    "KEYVAULT_URL" : "https://{keyvaultname}.vault.azure.net",
    "CMS_NON_SECURE_KEY_NAME": {key name from keyvault},
    "CMS_SECURE_KEY_NAME": {key name from keyvault },
    "API_VERSION": "2016-10-01"

```
#### Definition of done:
- [ ] Title of the pull request is clear and informative
- [ ] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session
- [ ] This change has appropriate test coverage
- [ ] Get at least two approvals

#### Reminders DURING merge
1. If you're merging from a short-term (feature) branch into a long-term branch (like dev, release, or master) then "Squash and merge" to keep our history clean.
1. If merging from two longterm branches (like cherry picks from upstream, dev to release, etc) then "Create merge commit" to preserve individual commits.

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 2 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 3 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 4 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

Configuration Pull Request
---
#### (For changes proposed to upstream)
Make sure that the following steps are done before merging

  - [ ] @devops team member has commented with :+1:
  - [ ] are you adding any new default values that need to be overridden when this goes live?  
    - [ ] Open a ticket (DEVOPS) to make sure that they have been added to secure vars.
    - [ ] Add an entry to the CHANGELOG.

### CODE REVIEW

File: playbooks/roles/edxapp/defaults/main.yml [hunk 1]
```diff
@@ -1455,6 +1466,13 @@ lms_env_config:
   GOOGLE_SITE_VERIFICATION_ID: "{{ EDXAPP_GOOGLE_SITE_VERIFICATION_ID }}"
   STATIC_URL_BASE: "{{ EDXAPP_LMS_STATIC_URL_BASE }}"
   LMS_ALLOWED_HOSTS: "{{ EDXAPP_LMS_ALLOWED_HOSTS }}"
+  GET_SECRETS_FROM_AZURE_KEYVAULT: "{{ EDXAPP_GET_SECRETS_FROM_AZURE_KEYVAULT }}"
+  AZURE_MSI_REQUEST_URI: "{{ EDXAPP_AZURE_MSI_REQUEST_URI }}" 
+  KEYVAULT_PAYLOAD: "{{ EDXAPP_KEYVAULT_PAYLOAD }}"
+  KEYVAULT_URL": "{{ EDXAPP_KEYVAULT_URL }}"
```
Comment (btelnes) [inline playbooks/roles/edxapp/defaults/main.yml]: Remove stray " before the colon on lines 1472-1475 and 1520-1523

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -1455,6 +1466,13 @@ lms_env_config:
...
+  LMS_NON_SECURE_KEY_NAME": "{{ EDXAPP_LMS_NON_SECURE_KEY_NAME }}"
+  LMS_SECURE_KEY_NAME": "{{ EDXAPP_LMS_SECURE_KEY_NAME }}"
+  API_VERSION": "{{ EDXAPP_API_VERSION }}"
```
Comment (btelnes) [inline playbooks/roles/edxapp/defaults/main.yml]: Suggest renaming as API_VERSION isn't specific enough.  Same comment for line 1523.

**Review by btelnes — CHANGES_REQUESTED** [review]

File: playbooks/roles/edxapp/defaults/main.yml (hunk 1 above)
Comment (manikarthikk) [reply playbooks/roles/edxapp/defaults/main.yml]: Updated

File: playbooks/roles/edxapp/defaults/main.yml (hunk 2 above)
Comment (manikarthikk) [reply playbooks/roles/edxapp/defaults/main.yml]: updated

**Review by btelnes — APPROVED** [review]

**Review by eltoncarr-ms — APPROVED** [review]

**Pull request merged**